### PR TITLE
Grafana UI: Add code variant to Text component

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Adds a monospaced code variant to make it available in the `Text` component. `code` is used as it aligns somewhat with the design tokens.

image

**Why do we need this feature?**

Sometimes we need to use a monospaced font.

**Who is this feature for?**

Developers

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Link/TextLink.tsx [hunk 1]
```diff
@@ -19,8 +19,8 @@ interface TextLinkProps extends Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 't
   external?: boolean;
   /** True when the link will be displayed inline with surrounding text, false if it will be displayed as a block. Depending on this prop correspondant default styles will be applied */
   inline?: boolean;
-  /** The default variant is 'body'. To fit another styles set the correspondent variant as it is necessary also to adjust the icon size */
-  variant?: keyof ThemeTypographyVariantTypes;
+  /** The default variant is 'body'. To fit another styles set the correspondent variant as it is necessary also to adjust the icon size. `code` is excluded, as it is not fit for links. */
+  variant?: keyof Omit<ThemeTypographyVariantTypes, 'code'>;
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Link/TextLink.tsx:23]: Just my initial thoughts, I can be convinced otherwise

Comment (JoaoSilvaGrafana) [reply packages/grafana-ui/src/components/Link/TextLink.tsx:23]: It seems like it makes sense, but just double checking with @staton-hyse11 what his opinion is. We can also remove this `Omit` later if we find a use case for it and it won't be a breaking change

File: packages/grafana-ui/src/components/Link/TextLink.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ interface TextLinkProps extends Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 't
 }
 
 const svgSizes: {
-  [key in keyof ThemeTypographyVariantTypes]: IconSize;
+  [key in keyof Omit<ThemeTypographyVariantTypes, 'code'>]: IconSize;
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Link/TextLink.tsx:32]: Could create a type at the top for this. Like `type TextLinkVariants = keyof Omit<ThemeTypographyVariantTypes, 'code'>;` and then we could use it both here and in line 23?

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Lgtm! Do we want to add something to the documentation both in storybook and [here](https://grafana.com/developers/saga/components/text) about how if developers want to display code they should use this variant?.

**Comment by tskarhed** [comment]
> Lgtm! Do we want to add something to the documentation both in storybook and [here](https://grafana.com/developers/saga/components/text) about how if developers want to display code they should use this variant?.

I see you point, but I do think we should make a code component (if we don't already have one?) for writing inline code that has some border and highlighting. So in general the `code` variant shouldn't be used, unless there are some rare cases? 🤔

**Comment by JoaoSilvaGrafana** [comment]
> > Lgtm! Do we want to add something to the documentation both in storybook and [here](https://grafana.com/developers/saga/components/text) about how if developers want to display code they should use this variant?.
> 
> I see you point, but I do think we should make a code component (if we don't already have one?) for writing inline code that has some border and highlighting. So in general the `code` variant shouldn't be used, unless there are some rare cases? 🤔
> 
> So maybe adding the code variant to `Don't use unless you are really sure that you should?`

Ah so wait do we want it to be a `code` variant or a more generic `monospace` variant? What's some examples where we would use this variant?

**Review by codeincarnate — APPROVED** [review]
Saw this and Dataviz could potentially use this so decided to check it out. Seems to work well!

**Comment by eledobleefe** [comment]
> > Lgtm! Do we want to add something to the documentation both in storybook and [here](https://grafana.com/developers/saga/components/text) about how if developers want to display code they should use this variant?.
> 
> I see you point, but I do think we should make a code component (if we don't already have one?) for writing inline code that has some border and highlighting. So in general the `code` variant shouldn't be used, unless there are some rare cases? 🤔
> 
> So maybe adding the code variant to `Don't use unless you are really sure that you should?`

Hi @tskarhed!
First of all, great job on this! 🙌 
My 2 cents here... I would also add some notes to the documentation. Devs will see that option and, unless we guide them, they can end up using this as we do not want. Maybe just add kind of what you said: "There is a 'code' variant that will show a monospace typography, but it should be used only when there are x and x requirements", for example 🤷 
Otherwise, LGTM 👌

**Review by eledobleefe — APPROVED** [review]

**Review by JoaoSilvaGrafana — APPROVED** [review]

**Pull request merged**